src/include/txn.i

```diff
@@ -856,21 +856,21 @@ __wt_txn_read(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt, WT_UPDATE *upd, WT
      * the current system.
      */
     if (__wt_txn_visible(session, start.txnid, start.timestamp)) {
-        WT_RET(__wt_update_alloc(session, &buf, &upd, &size, WT_UPDATE_STANDARD));
-        upd->txnid = start.txnid;
-        upd->start_ts = upd->durable_ts = start.timestamp;
-        F_SET(upd, WT_UPDATE_RESTORED_FROM_DISK);
+        WT_RET(__wt_update_alloc(session, &buf, updp, &size, WT_UPDATE_STANDARD));
+        (*updp)->txnid = start.txnid;
+        (*updp)->start_ts = start.timestamp;
+        F_SET((*updp), WT_UPDATE_RESTORED_FROM_DISK);
         *updp = upd;
```

Comment:
Won't this line overwrite everything you just set?

Reply:
Good point - I should've noticed!